### PR TITLE
added validation of schema before writing it to file

### PR DESCRIPTION
**Pull request by ihooldwork**

### CODE REVIEW

File: server/editor.go [hunk 1]
```diff
@@ -122,6 +122,10 @@ func setupEditor(server *Server) {
 				if existingSchema != nil {
 					return fmt.Errorf("ID has been taken")
 				}
+				_, err := schema.NewSchemaFromObj(context["resource"].(map[string]interface{}))
```
Comment (kwapik) [inline server/editor.go]: I think that `context["resource"].(map[string]interface{})` should be extracted to variable.

File: server/editor.go [hunk 2]
```diff
@@ -131,13 +135,18 @@ func setupEditor(server *Server) {
 					"message": context["response"],
 					"code":    201,
 				}
+
```
Comment (kwapik) [inline server/editor.go]: Unnecessary line

**Comment by kwapik** [comment]
Please also investigate failing tests, if this is caused by this PR fix it.

**Comment by nati** [comment]
Could you rebase commit?

**Comment by nati** [comment]
Could you rebase commit again? There was issue in code which got fixed.

**Comment by nati** [comment]
could you fix typo in commit message?

**Pull request merged**